universal_transfer_operator/src/data_providers/__init__.py

```diff
@@ -0,0 +1,43 @@
+import importlib
+import os.path
+
+from transfers.constants import TransferMode
+from transfers.data_providers.base import DataProviders, TempFile
+from transfers.datasets.base import UniversalDataset as Dataset
+
+from astro.constants import LoadExistStrategy
+from astro.utils.path import get_class_name
+
+DATASET_CONN_ID_TO_DATAPROVIDER_MAPPING = {
+    "s3": "data_providers.filesystem.aws.s3",
+    "aws": "data_providers.filesystem.aws.s3",
+    "gs": "data_providers.filesystem.google.cloud.gcs",
+    "google_cloud_platform": "data_providers.filesystem.google.cloud.gcs",
+}
+
+
+def create_dataprovider(
+    dataset: Dataset,
+    transfer_params: dict = {},
```

Comment:
We should never use mutable values as defaults. Use None instead and assign a default value within the function call if not provided by the user.

ref: https://medium.com/python-features/how-to-avoid-classic-pitfall-while-passing-default-values-in-python-7002c0dc4c7c

Reply:
[Done](https://github.com/astronomer/astro-sdk/pull/1492/commits/831d627d1a86acab300968fc56d11bc7f016b4b0)